homeassistant/helpers/entity_registry.py

```diff
@@ -324,41 +326,43 @@ def async_get_or_create(
         disabled_by: RegistryEntryDisabler | None = None,
         hidden_by: RegistryEntryHider | None = None,
         # Data that we want entry to have
-        area_id: str | None = None,
-        capabilities: Mapping[str, Any] | None = None,
-        config_entry: ConfigEntry | None = None,
-        device_id: str | None = None,
-        entity_category: EntityCategory | None = None,
-        has_entity_name: bool | None = None,
-        original_device_class: str | None = None,
-        original_icon: str | None = None,
-        original_name: str | None = None,
-        supported_features: int | None = None,
-        unit_of_measurement: str | None = None,
+        area_id: str | None | UndefinedType = UNDEFINED,
+        capabilities: Mapping[str, Any] | None | UndefinedType = UNDEFINED,
+        config_entry: ConfigEntry | None | UndefinedType = UNDEFINED,
+        device_id: str | None | UndefinedType = UNDEFINED,
+        entity_category: EntityCategory | UndefinedType | None = UNDEFINED,
+        has_entity_name: bool | UndefinedType = UNDEFINED,
+        original_device_class: str | None | UndefinedType = UNDEFINED,
+        original_icon: str | None | UndefinedType = UNDEFINED,
+        original_name: str | None | UndefinedType = UNDEFINED,
+        supported_features: int | None | UndefinedType = UNDEFINED,
+        unit_of_measurement: str | None | UndefinedType = UNDEFINED,
     ) -> RegistryEntry:
         """Get entity. Create if it doesn't exist."""
-        config_entry_id = None
-        if config_entry:
+        config_entry_id: str | None | UndefinedType = UNDEFINED
+        if not config_entry:
+            config_entry_id = None
+        elif config_entry is not UNDEFINED:
             config_entry_id = config_entry.entry_id
 
+        supported_features = supported_features or 0
+
         entity_id = self.async_get_entity_id(domain, platform, unique_id)
 
         if entity_id:
             return self.async_update_entity(
                 entity_id,
-                area_id=area_id or UNDEFINED,
-                capabilities=capabilities or UNDEFINED,
-                config_entry_id=config_entry_id or UNDEFINED,
-                device_id=device_id or UNDEFINED,
-                entity_category=entity_category or UNDEFINED,
-                has_entity_name=has_entity_name
-                if has_entity_name is not None
-                else UNDEFINED,
-                original_device_class=original_device_class or UNDEFINED,
-                original_icon=original_icon or UNDEFINED,
-                original_name=original_name or UNDEFINED,
-                supported_features=supported_features or UNDEFINED,
-                unit_of_measurement=unit_of_measurement or UNDEFINED,
+                area_id=area_id,
```

Comment:
Does this mean that we now allow implicitly updating an existing entity registry entry, resetting attributes to `None`, when using `async_get_or_create`? That feels a bit hidden and dangerous. :shrug:

Reply:
To clarify, I think it's ok to change default entity registry attributes here for the attributes where the user has separate override attributes possible and we don't change those. But resetting the same attribute implictly that the user may have changed doesn't feel right. I think there are some cases like that. I may be wrong.

Reply:
That's a good point @MartinHjelmare 👍 

`async_get_or_create` has this signature:
```py
    @callback
    def async_get_or_create(
        self,
        domain: str,
        platform: str,
        unique_id: str,
        *,
        # To influence entity ID generation
        known_object_ids: Iterable[str] | None = None,
        suggested_object_id: str | None = None,
        # To disable or hide an entity if it gets created
        disabled_by: RegistryEntryDisabler | None = None,
        hidden_by: RegistryEntryHider | None = None,
        # Data that we want entry to have
        area_id: str | None | UndefinedType = UNDEFINED,
        capabilities: Mapping[str, Any] | None | UndefinedType = UNDEFINED,
        config_entry: ConfigEntry | None | UndefinedType = UNDEFINED,
        device_id: str | None | UndefinedType = UNDEFINED,
        entity_category: EntityCategory | UndefinedType | None = UNDEFINED,
        has_entity_name: bool | UndefinedType = UNDEFINED,
        original_device_class: str | None | UndefinedType = UNDEFINED,
        original_icon: str | None | UndefinedType = UNDEFINED,
        original_name: str | None | UndefinedType = UNDEFINED,
        supported_features: int | None | UndefinedType = UNDEFINED,
        unit_of_measurement: str | None | UndefinedType = UNDEFINED,
    ) -> RegistryEntry:
```

Of concern are these parameters:
```py
        area_id: str | None | UndefinedType = UNDEFINED,  # I think we should remove this parameter!
        capabilities: Mapping[str, Any] | None | UndefinedType = UNDEFINED,  # Can't be set by user
        config_entry: ConfigEntry | None | UndefinedType = UNDEFINED,  # Can't be set by user
        device_id: str | None | UndefinedType = UNDEFINED,  # Can't be set by user
        entity_category: EntityCategory | UndefinedType | None = UNDEFINED,  # Can't be set by user
        has_entity_name: bool | UndefinedType = UNDEFINED,  # Can't be set by user
        original_device_class: str | None | UndefinedType = UNDEFINED,  # User sets separate attribute `device_class`
        original_icon: str | None | UndefinedType = UNDEFINED,  # User sets separate attribute `icon`
        original_name: str | None | UndefinedType = UNDEFINED,  # User sets separate attribute `name`
        supported_features: int | None | UndefinedType = UNDEFINED,  # Can't be set by user
        unit_of_measurement: str | None | UndefinedType = UNDEFINED,  # Can't be set by user
    ) -> RegistryEntry:
```

So, the potential concern is `area_id`.
There's no call to `async_get_or_create` which sets `area_id`, so I think we should just remove it.

Reply:
I think that we just at one point synced up the 2 method signatures

Reply:
But yeah let's drop anything that integrations shouldn't deal with.

Reply:
Sounds good.

Reply:
`area_id` parameter is removed in https://github.com/home-assistant/core/pull/77700